### PR TITLE
Add option to use function profile for composition in adiabatic conditions

### PR DESCRIPTION
**Pull request by gassmoeller**
See changes entry. I need this for an initial composition plugin that depends on the adiabatic conditions.

### CODE REVIEW

**Comment by gassmoeller** [comment]
Supersedes #1745

File: source/adiabatic_conditions/initial_profile.cc [hunk 1]
```diff
@@ -250,6 +260,72 @@ namespace aspect
...
+      prm.enter_subsection("Adiabatic conditions model");
+      {
+        prm.enter_subsection("Initial profile");
+        {
+          Functions::ParsedFunction<1>::declare_parameters (prm, 1);
+          prm.declare_entry("Composition reference profile","initial composition",
+                            Patterns::Selection("initial composition|function"),
+                            "Select how the reference profile for composition "
+                            "is computed. This profile is used to evaluate the "
+                            "material model, when computing the pressure, and "
```
Comment (tjhei) [inline source/adiabatic_conditions/initial_profile.cc]: remove comma before and?

File: tests/adiabatic_conditions_composition_function.prm [hunk 2]
```diff
@@ -0,0 +1,78 @@
+set Use years in output instead of seconds = false
```
Comment (tjhei) [inline tests/adiabatic_conditions_composition_function.prm]: can you leave a short comment here please?

**Comment by gassmoeller** [comment]
Fixed the comments, ready if the tester likes it.

**Pull request merged**